File: src/CSharp/GeometricObjects/Polygon.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Windows.Forms;
using System.Diagnostics;

using Kitware.VTK;

namespace ActiViz.Examples {
   public partial class Form1 : Form {
      public Form1() {
         InitializeComponent();
      }


      private void renderWindowControl1_Load(object sender, EventArgs e) {
         try {
            Polygon();
         }
         catch(Exception ex) {
            MessageBox.Show(ex.Message, "Exception", MessageBoxButtons.OK);
         }
      }


      private void Polygon() {
         // Setup four points
         vtkPoints points = vtkPoints.New();
         double c = Math.Cos(Math.PI / 6); // helper variable

         points.InsertNextPoint(0.0, -1.0, 0.0);
         points.InsertNextPoint(c, -0.5, 0.0);
         points.InsertNextPoint(c, 0.5, 0.0);
         points.InsertNextPoint(0.0, 1.0, 0.0);
         points.InsertNextPoint(-c, 0.5, 0.0);
         points.InsertNextPoint(-c, -0.5, 0.0);

         // Create the polygon
         vtkPolygon polygon = vtkPolygon.New();
         polygon.GetPointIds().SetNumberOfIds(6); //make a six-sided figure
         polygon.GetPointIds().SetId(0, 0);
         polygon.GetPointIds().SetId(1, 1);
         polygon.GetPointIds().SetId(2, 2);
         polygon.GetPointIds().SetId(3, 3);
         polygon.GetPointIds().SetId(4, 4);
         polygon.GetPointIds().SetId(5, 5);

         // Add the polygon to a list of polygons
         vtkCellArray polygons = vtkCellArray.New();
         polygons.InsertNextCell(polygon);

         // Create a PolyData
         vtkPolyData polygonPolyData = vtkPolyData.New();
         polygonPolyData.SetPoints(points);
         polygonPolyData.SetPolys(polygons);

         // Create a mapper and actor
         vtkPolyDataMapper mapper = vtkPolyDataMapper.New();
         mapper.SetInput(polygonPolyData);
         vtkActor actor = vtkActor.New();
         actor.SetMapper(mapper);
         vtkRenderWindow renderWindow = renderWindowControl1.RenderWindow;
         vtkRenderer renderer = renderWindow.GetRenderers().GetFirstRenderer();
         renderer.SetBackground(0.2, 0.3, 0.4);
         renderer.AddActor(actor);
         renderer.ResetCamera();
      }
   }
}

File: src/Cxx/Boneyard/GeometricObjects/WriteFile/Polygon.cxx
#include <vtkSmartPointer.h>
#include <vtkPolygon.h>
#include <vtkXMLPolyDataWriter.h>
#include <vtkCellArray.h>
#include <vtkPolyData.h>

int main(int, char *[])
{
  //Create four points
  vtkSmartPointer<vtkPoints> points = 
    vtkSmartPointer<vtkPoints>::New();
  points->InsertNextPoint(0.0, 0.0, 0.0);
  points->InsertNextPoint(1.0, 0.0, 0.0);
  points->InsertNextPoint(1.0, 1.0, 0.0);
  points->InsertNextPoint(0.0, 1.0, 0.0);

  //Create a polygon
  vtkSmartPointer<vtkPolygon> polygon = 
    vtkSmartPointer<vtkPolygon>::New();
  polygon->GetPointIds()->SetNumberOfIds(4); //make a quad
  polygon->GetPointIds()->SetId(0, 0);
  polygon->GetPointIds()->SetId(1, 1);
  polygon->GetPointIds()->SetId(2, 2);
  polygon->GetPointIds()->SetId(3, 3);
  //this is an unfortunate example, as you cannot tell which 0,1,2 or 3 is which!
  //A clearer statement is SetId(0, IndexOfPoint0)

  //Create a cell array to add the polygon to.
  vtkSmartPointer<vtkCellArray> polygons = vtkSmartPointer<vtkCellArray>::New();
  polygons->InsertNextCell(polygon);

  //Add the points and polygon to a polydata object.
  vtkSmartPointer<vtkPolyData> polydata = vtkSmartPointer<vtkPolyData>::New();
  polydata->SetPoints(points);
  polydata->SetPolys(polygons);

  //Write the file.
  vtkSmartPointer<vtkXMLPolyDataWriter> writer = 
    vtkSmartPointer<vtkXMLPolyDataWriter>::New();
  writer->SetFileName("Polygon.vtp");
  writer->SetInput(polydata);
  writer->Write();

  return EXIT_SUCCESS;
}
